lightning-graph-sync/src/processing.rs

```diff
@@ -0,0 +1,434 @@
+			synthetic_update.fee_base_msat = fee_base_msat;
+		}
+
+		if channel_flags & 0b_0000_1000 > 0 {
+			let fee_proportional_millionths: u32 = Readable::read(read_cursor)?;
+			synthetic_update.fee_proportional_millionths = fee_proportional_millionths;
+		}
+
+		if channel_flags & 0b_0000_0100 > 0 {
+			let tentative_htlc_maximum_msat: u64 = Readable::read(read_cursor)?;
+			synthetic_update.htlc_maximum_msat = if tentative_htlc_maximum_msat == u64::max_value()
+			{
+				OptionalField::Absent
+			} else {
+				OptionalField::Present(tentative_htlc_maximum_msat)
+			};
+		}
+
+		network_graph.update_channel_unsigned(&synthetic_update)?;
+	}
+
+	let input_termination_check: Result<u8, DecodeError> = Readable::read(read_cursor);
+	if let Err(DecodeError::ShortRead) = input_termination_check {
+		Ok(())
+	} else {
+		Err(GraphSyncError::ProcessingError(
+			"unexpected data found".to_string(),
+		))
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use std::thread;
+	use std::time::Duration;
+
+	use bitcoin::blockdata::constants::genesis_block;
+	use bitcoin::Network;
+
+	use lightning::ln::msgs::DecodeError;
+	use lightning::routing::network_graph::NetworkGraph;
+
+	use crate::error::GraphSyncError;
+	use crate::processing::update_network_graph;
+
+	#[test]
+	fn network_graph_fails_to_update_from_clipped_input() {
+		let block_hash = genesis_block(Network::Bitcoin).block_hash();
+		let network_graph = NetworkGraph::new(block_hash);
+
+		// OldestDataDirection0, recent timestamp, with last byte removed
+		let example_input = vec![76, 68, 75, 1, 111, 226, 140, 10, 182, 241, 179, 114, 193, 166, 162, 70, 174, 99, 247, 79, 147, 30, 131, 101, 225, 90, 8, 156, 104, 214, 25, 0, 0, 0, 0, 0, 97, 227, 98, 218, 0, 0, 0, 4, 2, 22, 7, 207, 206, 25, 164, 197, 231, 230, 231, 56, 102, 61, 250, 251, 187, 172, 38, 46, 79, 247, 108, 44, 155, 48, 219, 238, 252, 53, 192, 6, 67, 2, 36, 125, 157, 176, 223, 175, 234, 116, 94, 248, 201, 225, 97, 235, 50, 47, 115, 172, 63, 136, 88, 216, 115, 11, 111, 217, 114, 84, 116, 124, 231, 107, 2, 158, 1, 242, 121, 152, 106, 204, 131, 186, 35, 93, 70, 216, 10, 237, 224, 183, 89, 95, 65, 3, 83, 185, 58, 138, 181, 64, 187, 103, 127, 68, 50, 2, 201, 19, 17, 138, 136, 149, 185, 226, 156, 137, 175, 110, 32, 237, 0, 217, 90, 31, 100, 228, 149, 46, 219, 175, 168, 77, 4, 143, 38, 128, 76, 97, 0, 0, 0, 2, 0, 0, 255, 8, 153, 192, 0, 2, 27, 0, 0, 0, 1, 0, 0, 255, 2, 68, 226, 0, 6, 11, 0, 1, 2, 3, 0, 0, 0, 2, 0, 40, 0, 0, 0, 0, 0, 0, 3, 232, 0, 0, 0, 100, 0, 0, 2, 224, 0, 0, 0, 0, 29, 129, 25, 192, 255, 8, 153, 192, 0, 2, 27, 0, 0, 36, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 58, 85, 116, 216, 255, 2, 68, 226, 0, 6, 11, 0, 1, 24, 0, 0, 3, 232, 0, 0, 0];
+		let update_result = update_network_graph(&network_graph, &example_input[..]);
+		assert!(update_result.is_err());
+		if let Err(GraphSyncError::DecodeError(DecodeError::ShortRead)) = update_result {
+			// this is the expected error type
+		} else {
+			assert!(false);
```

Comment:
Would it be simple to panic with the `Debug` representation of the unexpected result?

Reply:
I can implement a Debug for GraphSyncError if that's what you're asking? I haven't so far because it's merely a proxy for other error types, lmk.

Reply:
Yeah, just so you can have a better failure message:

```rust
panic!("Unexpected result: {:?}", update_result);
```